feat: /internet-of-things/appstore redesgin

mtruj013 opened the pull request:
## Done

- Applied redesign as per [doc](https://docs.google.com/document/d/1CKwmPDDtXOIgw-jSZFwzP-qLMc_rS_dUACpHQXlg0fA/edit?tab=t.0) and [design](https://www.figma.com/design/1CdIT6TV3ASf55GsnQcwXg/Internet-of-Things?node-id=0-1&node-type=canvas&m=dev)

## QA

- View the site locally in your web browser at: https://ubuntu-com-14523.demos.haus/internet-of-things/appstore
    - Be sure to test on mobile, tablet and desktop screen sizes
- Check against doc and design

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-16630

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14523.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/5402)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

eliman11 commented:
Thanks @mtruj013! Looks good to me, some changes to change to US spelling - I've marked them on the copydoc too if it's easier to spot that way. Feel free to add UX+1 once these are changed.
- [ ] 'Specialized' under 'Ideal for secure software distribution'
- [ ] 'Minimize' under 'Reliable over-the-air (OTA) updates' (subheading)
- [ ] 'Maximize' under 'Maximize bandwidth and speed by caching downloads and snap traffic.'
- [ ] 'Specializing' under 'IoT Professional Services'

Also some of the sections are missing dividers but will leave that with @mattea-turic :)
<img width="1433" alt="Screenshot 2024-12-02 at 17 40 24" src="https://github.com/user-attachments/assets/3a03a2d7-c817-4c8b-80dd-c371e8ff3eb0">

mattea-turic commented:
ty @mtruj013 ! a few comments from me:

for this section:
<img width="500" alt="Screenshot 2024-12-03 at 11 17 11" src="https://github.com/user-attachments/assets/69766818-a9be-428d-9ad1-008e4a0c44a4">
- [ ] hrs between the list items pls

for this section:
<img width="1443" alt="Screenshot 2024-12-03 at 11 18 25" src="https://github.com/user-attachments/assets/311d62bc-f0f7-4d09-a7a3-094cf1122081">
- [ ] muted hr above the equal height row

for this section:
<img width="500" alt="Screenshot 2024-12-03 at 11 20 58" src="https://github.com/user-attachments/assets/7aab24c6-6351-4c1a-8530-d317c87b4bd9">
- [ ] dividers between the list items, incl. above the first one
- [ ] spacing is missing between the text, "ready to find out more" and the button

for this section:
<img width="500" alt="Screenshot 2024-12-03 at 11 23 57" src="https://github.com/user-attachments/assets/dd0fc554-d102-4b80-abb6-e607273f58c4">
- [ ]  hrs missing at the bottom of each container (see figma maybe since i'm not sure i'm describing this well)

for this section:
<img width="500" alt="Screenshot 2024-12-03 at 11 28 12" src="https://github.com/user-attachments/assets/1f8bbbc2-9ecf-4196-bf80-113057aa7167">
- [ ] the first quote seems to have more bottom padding than the second one. Could you make it so that they're each wrapped in just a shallow?
On mobile, it might be worth to omit the images as they end up taking too much space (unless this can be swapped with a size more compatible with small screens; I can provide the assets though can you switch between variants like that depending on breakpoint?):
<img width="380" alt="Screenshot 2024-12-03 at 11 35 57" src="https://github.com/user-attachments/assets/7ea49d23-c809-415c-bf5f-158faad2d9a2">

And for this section (mobile):
<img width="500" alt="Screenshot 2024-12-03 at 11 39 38" src="https://github.com/user-attachments/assets/3bb99473-ec14-495f-901f-a525413b6ca9">
- [ ]  is it possible to minimise the spacing between the content in the card and the link, since the cards are stacked vertically, so they don't have to occupy the same length (unlike on desktop)

mtruj013 commented:
Thanks @eliman11 and @mattea-turic ! I've addressed most of the comments with the exception of the following:

> hrs missing at the bottom of each container (see figma maybe since i'm not sure i'm describing this well) 

Sorry, I'm not seeing what you mean. The demo seems to match the figma:
demo:
![image](https://github.com/user-attachments/assets/7d4d1249-5582-48a3-951d-a48b092ff4d4)

figma:
![image](https://github.com/user-attachments/assets/4d7cfc2c-170a-459d-bf57-50e687f18ba9)

Not including hrs for each link is also consistent with how we use this design elsewhere, examples [here](https://canonical.com/data/valkey) and [here](https://canonical.com/data/kafka/support).

> On mobile, it might be worth to omit the images as they end up taking too much space (unless this can be swapped with a size more compatible with small screens; I can provide the assets though can you switch between variants like that depending on breakpoint?

Yes please, the assets currently have a lot of white space baked in so reducing the size makes the text illegible on smaller screens:
![image](https://github.com/user-attachments/assets/9456174a-44c8-4ddd-80b5-0d34584f0f8f)

Should have no/very minimal whitespace to format correctly, eg from u.com/aws:
![image](https://github.com/user-attachments/assets/addcfeb6-8feb-420a-ae43-12f9206587ac)

mattea-turic commented:
ty @mtruj013 ! Here are the assets for [screenly](https://assets.ubuntu.com/v1/3cd4128e-screenly.png), [domotz](https://assets.ubuntu.com/v1/3c4beadc-domotz.png), and [rigado](https://assets.ubuntu.com/v1/e512541f-rigado.png)

eliman11 commented:
Looks good, thank you! Adding +1

mattea-turic commented:
Looks great, thanks @mtruj013 ! I'll add a +1;
there's one v small detail that'll need to be changed, but I think this is a super minor fix, where there's an extra hr above "onboard new devices" on smaller screens
<img width="668" alt="Screenshot 2024-12-04 at 09 50 07" src="https://github.com/user-attachments/assets/406a707d-e71f-4c0e-b362-d636d2aad0d6">

immortalcodes commented:
![image](https://github.com/user-attachments/assets/79382b0a-14cd-4dff-a923-82334682845c)
![image](https://github.com/user-attachments/assets/5ea43d64-6f8f-4dea-a6b4-04b68c2c9838)

Design is a bit different here, but I guess it is already talked out?

mattea-turic commented:
> Design is a bit different here, but I guess it is already talked out?

Ah I don't remember that changing on the figma; the demo is how I had initially intended it, so all good there

immortalcodes commented:
![image](https://github.com/user-attachments/assets/6531e504-86bc-44cd-a5b3-899ae567293f)
Double lines on small screen

mtruj013 commented on templates/internet-of-things/appstore.html:
```diff
@@ -112,238 +132,355 @@ <h2>Check out the dedicated Snap Store</h2>
+          <div class="p-image-container is-highlighted">
+            {{ image(url="https://assets.ubuntu.com/v1/54e5c806-dell-technologies.png",
+                        alt="",
+                        width="852",
+                        height="481",
```
Nah, that's what the aria-label on the link that's wrapping the image is for 🙂 Original ux/a11y feedback for this pattern [here](https://github.com/canonical/canonical.com/pull/1386#issuecomment-2397136348)
